tests/system/src/test/java/test/javafx/scene/control/SystemMenuBarTest.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright (c) 2023, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+package test.javafx.scene.control;
+
+import com.sun.javafx.tk.Toolkit;
+import javafx.event.ActionEvent;
+import javafx.event.EventHandler;
+import javafx.scene.Scene;
+import javafx.scene.control.Menu;
+import javafx.scene.control.MenuBar;
+import javafx.scene.control.MenuItem;
+import javafx.scene.layout.VBox;
+import javafx.stage.Stage;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.AfterAll;
+import test.com.sun.javafx.pgstub.StubToolkit;
+import test.util.Util;
+import test.util.memory.JMemoryBuddy;
+import javafx.application.Platform;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+
+import static org.junit.Assert.assertTrue;
+
+public class SystemMenuBarTest {
+
+    @BeforeAll
+    public static void initFX() throws Exception {
+        CountDownLatch startupLatch = new CountDownLatch(1);
+        Platform.setImplicitExit(false);
+        Util.startup(startupLatch, startupLatch::countDown);
+    }
+
+    @AfterAll
+    public static void teardownOnce() {
+        Util.shutdown();
+    }
+
+    @Test
+    public void test_JDK_8299423() {
+        /**
+         * We want to test, if we add a menu to the system-menu, whether it's callable get's collected after the entry is removed.
+         *
+         */
+        JMemoryBuddy.memoryTest((checker) -> {
```

Comment:
Thank you for explaining what the test attempts to do!

minor: move javadoc above the `@Test`, and possibly rephrase the description, maybe something like this:

"Tests whether the ActionListener associated with the system menu gets collected after said menu is removed."